src/connector/src/source/kafka/source/reader.rs

```diff
@@ -194,6 +195,14 @@ impl CommonSplitReader for KafkaSplitReader {
         let mut num_messages = 0;
         let max_chunk_size = self.source_ctx.source_ctrl_opts.chunk_size;
         let mut res = Vec::with_capacity(max_chunk_size);
+        // ingest kafka message header can be expensive, do it only when required
+        let require_message_header = self
+            .parser_config
+            .common
+            .rw_columns
+            .iter()
+            .any(|col_desc| col_desc.additional_column_type == AdditionalColumnType::Header);
```

Comment:
Will we do this check for other columns?

Reply:
no, we always ingest timestamp and key.
for key, some formats take it as default, eg. debezium and upsert. Although it can be configured to ingest or not, I prefer not doing it in this pr.
for timestamp, it is just an int and has little overhead.

However, the given header in Kafka SDK is a borrowed one, copying can be expensive.